app/components/dynamic-select/dynamic-select.tsx

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+import { useModelFilters } from "~/hooks";
+import type { ModelFilterItem, ModelFilterProps } from "~/hooks";
+import { tw } from "~/utils";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../forms";
+import Input from "../forms/input";
+import { Button } from "../shared";
+import type { Icon } from "../shared/icons-map";
+import When from "../when/when";
+
+type Props = ModelFilterProps & {
+  className?: string;
+  style?: React.CSSProperties;
+  label?: React.ReactNode;
+  searchIcon?: Icon;
+  showSearch?: boolean;
+  defaultValue?: string;
+  renderItem?: (item: ModelFilterItem) => React.ReactNode;
+  extraContent?: React.ReactNode;
+};
+
+export default function DynamicSelect({
+  className,
+  style,
+  label,
+  searchIcon = "search",
+  showSearch = true,
+  defaultValue,
+  model,
+  countKey,
+  initialDataKey,
+  renderItem,
+  extraContent,
+}: Props) {
+  const {
+    searchQuery,
+    setSearchQuery,
+    handleSearchQueryChange,
+    items,
+    totalItems,
+    clearFilters,
+    selectedItems,
+    resetModelFiltersFetcher,
+    handleSelectItemChange,
+  } = useModelFilters({
```

Comment:
This should be changed, right?

Reply:
We have to do this when we are in update... suppose you selected an item after searching... so that item will not come directly in `initialItems`... so we need to have something to get the selected item initially

Reply:
Hey man. I think we have a confusion.
I am talking about the `asset.new` and `asset.edit` where we use the `DynamicSelect`.

1. DynamicSelect should not push to search params when its value changes
2. The default value for edit comes from the asset itself. On edit it gets passed to `AssetForm`